wandb/sdk/wandb_artifacts.py

```diff
@@ -1465,22 +1486,41 @@ def _entry_from_obj(
         prefix: str = "",
         multi: bool = False,
     ) -> ArtifactManifestEntry:
-        ref = path
+        """
+        Arguments:
+            obj: The S3 object
+            path: The S3-style path (e.g.: "s3://bucket/file.txt")
+            name: The user assigned name, or None if not specified
+            prefix: The prefix to add (will be the same as `path` for directories)
+            multi: Whether or not this is a multi-object add
+        """
+        bucket, key, _ = self._parse_uri(path)
+
+        # Always use posix paths, since that's what S3 uses.
+        posix_key = pathlib.PurePosixPath(obj.key)  # the bucket key
+        posix_path = pathlib.PurePosixPath(bucket) / pathlib.PurePosixPath(
+            key
+        )  # the path, with the scheme stripped
+        posix_prefix = pathlib.PurePosixPath(prefix)  # the prefix, if adding a prefix
+        posix_name = pathlib.PurePosixPath(name or "")
+        posix_ref = posix_path
```

Comment:
nit: should this one be in the else condition?

Reply:
Ah you mean explicitly add this as part of the `else` right after the `elif multi` case? That might be cleaner.